Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -422,46 +422,66 @@ extension SwiftLanguageServer {
     cancelInFlightPublishDiagnosticsTask(for: note.textDocument.uri)
 
     let keys = self.keys
-    var edits: [IncrementalEdit] = []
+    struct Edit {
+      let offset: Int
+      let length: Int
+      let replacement: String
+    }
+
+    var edits: [Edit] = []
 
     let editResult = self.documentManager.edit(note) {
```

Comment:
I haven't looked at this too much, but is the callback from `edit` even really needed any more? Or should we just do the TextDocumentContentChangeEvent -> [Edit] outside it?

Reply:
It is, because the callback is transforming the concurrent edits from the LSP request into sequential edits that we can send to sourcekitd.

Reply:
Okay, it could also just return those rather than run the callback. It would mean another top level `Edit` though, which we don't have enough of :P (Edit/SourceEdit/IncrementalEdit all in swift-syntax).

Reply:
Filed https://github.com/apple/sourcekit-lsp/issues/939